src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/TracePoint.kt

```diff
@@ -172,47 +182,47 @@ internal class MonitorEnterTracePoint(
     callStackTrace: CallStackTrace,
     stackTraceElement: StackTraceElement
 ) : CodeLocationTracePoint(iThread, actorId, callStackTrace, stackTraceElement) {
-    override fun toStringImpl(): String = "MONITORENTER at " + stackTraceElement.shorten()
+    override fun toStringImpl(withLocation: Boolean): String = "MONITORENTER at " + stackTraceElement.shorten()
```

Comment:
Let's create an abstract method for the first part only (such as `MONITORENTER`) here. It will prevent you from having bugs like here -- you are ignoring the `withLocation` property. With the suggestion, this logic of adding the location part will be written once in the parent class only.